modules/record_collection_data/main.tf

```diff
@@ -0,0 +1,13 @@
+resource "cloudflare_record" "this_data" {
```

Comment:
You have to iterate here in the same way as in `record_collection` submodule.

Reply:
Why is that? The module's iterative invocation happens in the main file.

Reply:
See my comments to `main.tf`